crates/grit-pattern-matcher/src/pattern/regex.rs

```diff
@@ -93,16 +93,13 @@ impl<Q: QueryContext> RegexPattern<Q> {
                     continue;
                 }
             } else {
-                let res = if let Some((Some(mut position), Some(source))) = binding
+                let res = if let Some((Some(mut byte_range), Some(source))) = binding
                     .get_last_binding()
-                    .map(|binding| (binding.position(context.language()), binding.source()))
+                    .map(|binding| (binding.range(context.language()), binding.source()))
                 {
-                    // this moves the byte-range out of sync with
-                    // the row-col range, maybe we should just
-                    // have a Range<usize> for String bindings?
-                    position.end_byte = position.start_byte + range.end as u32;
-                    position.start_byte += range.start as u32;
-                    Q::ResolvedPattern::from_range_binding(position, source)
```

Comment:
Here is a gotcha from the previous implementation that won't be a concern anymore.